soh/soh/Enhancements/debugconsole.cpp

```diff
@@ -31,18 +29,22 @@ extern PlayState* gPlayState;
 }
 
 #include <libultraship/bridge.h>
+#include <libultraship/libultraship.h>
 
-#define CMD_REGISTER LUS::GetConsole()->AddCommand
+#define CMD_REGISTER LUS::Context::GetInstance()->GetConsole()->AddCommand
+// TODO: Commands should be using the output passed in.
+#define ERR_MESSAGE std::reinterpret_pointer_cast<LUS::ConsoleWindow>(LUS::Context::GetInstance()->GetWindow()->GetGui()->GetGuiWindow("Console"))->SendErrorMessage
+#define INF_MESSAGE std::reinterpret_pointer_cast<LUS::ConsoleWindow>(LUS::Context::GetInstance()->GetWindow()->GetGui()->GetGuiWindow("Console"))->SendInfoMessage
 
-static bool ActorSpawnHandler(std::shared_ptr<LUS::Console> Console, const std::vector<std::string>& args) {
+static bool ActorSpawnHandler(std::shared_ptr<LUS::Console> Console, const std::vector<std::string>& args, std::string* output) {
     if ((args.size() != 9) && (args.size() != 3) && (args.size() != 6)) {
-        LUS::GetConsole()->SendErrorMessage("Not enough arguments passed to actorspawn");
-        return CMD_FAILED;
+        ERR_MESSAGE("Not enough arguments passed to actorspawn");
+        return 1;
```

Comment:
Seems like a net-negative to this file to move from enums to explicit numbers, was there a reason for this?

Reply:
@Kenix3 looks like this happened here https://github.com/HarbourMasters/Shipwright/commit/40bfce61ad9e29e90b6811f6902d2f5a10564abe

do you know the story behind this change?

Reply:
I don't really like a macro that is just true or false.
Beyond that, these commands should be returning a failure code or success (0). Remember that the return value was changed to mimic how command line return codes are used.